src/main/java/org/terasology/metalrenegades/world/dynamic/discoverables/DiscoverablesProvider.java

```diff
@@ -33,8 +34,10 @@ public void setSeed(long seed) {
 
     @Override
     public void process(GeneratingRegion region) {
+        Border3D border = region.getBorderForFacet(DiscoverablesFacet.class).extendBy(0, 0, 10);
+
         SurfaceHeightFacet surfaceHeightFacet = region.getRegionFacet(SurfaceHeightFacet.class);
-        DiscoverablesFacet facet = new DiscoverablesFacet(region.getRegion(), region.getBorderForFacet(DiscoverablesFacet.class));
+        DiscoverablesFacet facet = new DiscoverablesFacet(region.getRegion(), border);
 
         Region3i worldRegion = facet.getWorldRegion();
```

Comment:
`Region3i` is deprecated. Use `BlockRegion` instead.

Reply:
I swapped out the `Region3i` statement for this instead:
`Rect2i worldRegion = surfaceHeightFacet.getWorldRegion();`

This alternative was taken directly from other similar world providers as a way to get the current world region. However, this causes world generation to crash with errors similar to the following:
`[Chunk-Generator-7] ERROR o.t.u.concurrency.TaskProcessor - Error in thread Chunk-Generator-7
java.lang.IllegalArgumentException: Out of bounds: (-37, 11, 45) for region (Min: (-33, 0, 31), Size: (34, 65, 34))
	at org.terasology.world.generation.facets.base.SparseFacet3D.checkWorldCoords(SparseFacet3D.java:56)
	at org.terasology.world.generation.facets.base.SparseObjectFacet3D.setWorld(SparseObjectFacet3D.java:91)
	at org.terasology.metalrenegades.world.dynamic.discoverables.DiscoverablesProvider.process(DiscoverablesProvider.java:50)
	at org.terasology.world.generation.RegionImpl.lambda$getFacet$1(RegionImpl.java:49)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.terasology.world.generation.RegionImpl.getFacet(RegionImpl.java:48)
	at org.terasology.metalrenegades.world.dynamic.discoverables.DiscoverablesRasterizer.generateChunk(DiscoverablesRasterizer.java:41)
	at org.terasology.world.generation.WorldImpl.rasterizeChunk(WorldImpl.java:64)
	at org.terasology.world.generation.BaseFacetedWorldGenerator.createChunk(BaseFacetedWorldGenerator.java:75)
	at org.terasology.world.chunks.localChunkProvider.LocalChunkProvider$1.run(LocalChunkProvider.java:649)
	at org.terasology.utilities.concurrency.TaskProcessor.run(TaskProcessor.java:50)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)`

I don't know why these errors appear, since switching out `Region3i` for `Rect2i` should result in the same region, but the provider is somehow generating positions that are out of bounds.

Reply:
I found a way to fix this issue, it should all be good now 👍